### PR TITLE
Feat(front): Enable Sort and Filter editing

### LINKED ISSUE #956: Enable Sort and Filter editing
# Current Behavior

Today, users can't change the active sort and filter fields by clicking on the blue field. This makes it hard to modify the sorting and filtering quickly and easily. (you need to delete the existing one to modify it)

https://user-images.githubusercontent.com/102394218/256372062-209a7135-187e-4ce5-a5e1-8b377c4e90dc.mp4

# Desired Behavior

**Check #1215 for updated style changes first**

Users should be able to easily edit the active sort and filter by clicking on it.

List of changes on the active sort and filter:
- Clickable: Enable clickability with hover effects - background-color changes to Accent/tertiary and border to Accent/primary.
- Change the hover effect of the tabler-icon-x (delete button) to Accent/secondary background-color.
- Disable text selection when clicking on the element.

**The style change should also be applied to the opportunities filters**

### Sort

When clicking on the blue field of the active sort, it should open the same menu as when you click on the "Sort" button, allowing users to change the sorting parameters (e.g., name, employees).

https://user-images.githubusercontent.com/102394218/256371039-2f8f23e8-7381-45c2-96f6-b30e4be047f5.mp4

### Filter

Similarly, when clicking on a filter, it should open the filter menu of the selected category, giving users the option to modify the entered value (e.g., in Account owner: is Tim Cook, allow change of "Tim Cook").

https://user-images.githubusercontent.com/102394218/256371053-3fdb1e74-c58e-4b3b-92a1-b44c918aa1e8.mp4

https://www.figma.com/file/xt8O9mFeLl46C5InWwoMrN/Twenty?type=design&node-id=6775-72724&mode=design&t=JHNORCQzqa5Ms34q-4

### PR DESCRIPTION
**Pull request by gitstart-twenty**
## Demo: 

https://github.com/twentyhq/twenty/assets/140154534/af98067f-7572-4957-b3de-ca4415cb459e

Fixes #956

### CODE REVIEW

File: front/src/modules/ui/view-bar/components/MultipleFiltersDropdownButton.tsx [hunk 1]
```diff
@@ -8,17 +10,35 @@ import { ViewBarDropdownButton } from './ViewBarDropdownButton';
...
   return (
-    <ViewBarDropdownButton
-      dropdownId={FilterDropdownId}
-      buttonComponent={<MultipleFiltersButton />}
-      dropdownComponents={<MultipleFiltersDropdownContent />}
-      dropdownHotkeyScope={hotkeyScope}
-    />
+    <StyledDropdownContainer isInViewBar={isInViewBar}>
+      <ViewBarDropdownButton
+        dropdownId={isInViewBar ? hotkeyScope.scope : FilterDropdownId}
```
Comment (lucasbordeau) [inline front/src/modules/ui/view-bar/components/MultipleFiltersDropdownButton.tsx:35]: Not sure if using a hotkeyScope as a dropdownId is relevant, could we create another dropdown id ? Do we need a dynamic one ? We could add a new props dropdownId in `MultipleFiltersDropdownButton` as a default way of doing this.

**Review by lucasbordeau — COMMENTED** [review]
This solution works but I feel like it could be refactored, we'll handle it in https://github.com/twentyhq/twenty/issues/1873

File: front/src/modules/ui/view-bar/components/MultipleFiltersDropdownButton.tsx (hunk 1 above)
Comment (gitstart-twenty) [reply front/src/modules/ui/view-bar/components/MultipleFiltersDropdownButton.tsx:35]: Let us look into this @lucasbordeau

**Comment by gitstart-twenty** [comment]
Hey @lucasbordeau ,
We made some changes here

**Comment by lucasbordeau** [comment]
I'll let @charlesBochet review this !

**Comment by charlesBochet** [comment]
@gitstart-twenty We have refactored this part of the code a lot, could you take another shot based on main branch?

**Comment by gitstart-twenty** [comment]
Hey @charlesBochet,
What do you think of this approach?

**Comment by charlesBochet** [comment]
@gitstart-twenty Looks better! I still have buggy behaviors while editing a filter:

Missing values:
image

image

**Comment by gitstart-twenty** [comment]
Hey @charlesBochet Please give this another look when you get a chance

**Comment by charlesBochet** [comment]
@gitstart-twenty As discussed during our meeting, we are not going to merge this one. We have discussed it internally and we will be going with the following implementation: https://github.com/twentyhq/twenty/pull/2968

You can mark this as complete on your end, the work has been done, sorry for the change in the direction

**Pull request closed without merging**